Source/HBIOS/kbd.asm

```diff
@@ -59,14 +66,29 @@ KBD_IDLE	.DB	0	; IDLE COUNT
 	DEVECHO	"KBD: ENABLED\n"
 ;
 ;__________________________________________________________________________________________________
+; HARDWARE LEVEL INTERFACE
+;__________________________________________________________________________________________________
+;
+#INCLUDE "ps2iface.inc"
+;__________________________________________________________________________________________________
 ; KEYBOARD INITIALIZATION
 ;__________________________________________________________________________________________________
 ;
 KBD_INIT:
+#IF ((INTMODE == 2) & INTPS2KBD)
+	CALL	KBDQINIT		; INITIALIZE QUEUE
+	LD	HL,KBD_INT		; INSTALL VECTOR
+	LD	(IVT(INT_PS2KB)),HL	; IVT INDEX

```

Comment:
I don't think the interrupt vector should be installed until after you have confirmed the existence of the keyboard controller.

Reply:
Thanks Wayne I think I understand what I need to fix this.